src/timepicker/timepicker.js

```diff
@@ -55,6 +55,46 @@ angular.module('ui.bootstrap.timepicker', [])
     });
   }
 
+  var min;
+  $scope.$parent.$watch($parse($attrs.min), function(value) {
+    var dt = new Date(value);
+    min = isNaN(dt) ? undefined : dt;
+  });
+
+  var max;
+  $scope.$parent.$watch($parse($attrs.max), function(value) {
+    var dt = new Date(value);
+    max = isNaN(dt) ? undefined : dt;
+  });
+
+  $scope.noIncrementHours = function() {
+    var incrementedSelected = addMinutes( selected, hourStep * 60 );
+    return incrementedSelected > max || (min && incrementedSelected < selected);
```

Comment:
So what I meant was more `return (max && incrementedSelected > max) || incrementedSelected < selected;`, and so on.